apps/studio/src/features/editing-experience/components/LayoutSelection.tsx

```diff
@@ -0,0 +1,281 @@
+  return (
+    <>
+      {/* This is the floating header */}
+      <VStack position="fixed" h="100%">
+        <VStack p="1.5rem 2rem" w="full" gap="0.5rem" alignItems="flex-start">
+          <Button variant="link">
+            <Icon as={BiLeftArrowAlt} />
+            Back to all pages
+          </Button>
+          <HStack p="0" justifyContent="space-between" w="100%">
+            <Text textStyle="h4" color="base.content.default">
+              {`Choose a layout for "`}
+              {props.pageName}
+              {`"`}
+            </Text>
+            <Button size="xs">
+              Start with this layout
+              <BiRightArrowAlt fontSize="1.5rem" />
+            </Button>
+          </HStack>
+        </VStack>
+        <Grid
+          w="100%"
+          h="calc(100% - 7rem)"
+          templateColumns="repeat(4, 1fr)"
+          gap={0}
+        >
+          <GridItem w="100%" colSpan={1} bg="grey.50" overflowY="scroll">
+            <VStack p="2rem" gap="2rem" h="fit-content">
+              {/* Picking Layouts */}
+              {LAYOUT_DATA.map((layoutEntry) => {
+                return (
+                  <VStack
+                    key={layoutEntry.layoutTypename}
+                    p="0"
+                    gap="0"
+                    cursor="pointer"
+                    onClick={() => {
+                      setSelectedLayout(layoutEntry)
+                    }}
+                    role="group"
+                  >
+                    <Box
+                      className="hover-border"
+                      borderRadius="8px"
+                      border="2px"
+                      p="1.5rem 1.5rem 0 1.5rem"
+                      mb="1.25rem"
+                      position="relative"
+                      borderColor={
+                        selectedLayout.layoutTypename ===
+                        layoutEntry.layoutTypename
+                          ? 'base.divider.brand'
+                          : 'base.divider.medium'
+                      }
+                      bgColor={
+                        selectedLayout.layoutTypename ===
+                        layoutEntry.layoutTypename
+                          ? 'interaction.muted.main.active'
+                          : 'interaction.muted.main.hover'
+                      }
+                      sx={
+                        selectedLayout.layoutTypename !==
+                        layoutEntry.layoutTypename
+                          ? {
+                              transition:
+                                'border-color 300ms ease-out, opacity 300ms ease-out, background-color 300ms ease-out',
+                            }
+                          : {}
+                      }
+                      _groupHover={
+                        selectedLayout.layoutTypename !==
+                        layoutEntry.layoutTypename
+                          ? {
+                              borderColor: useToken(
```

Comment:
any reason why `useToken` was used here too? should be able to use the tokens directly. Should also have intellisense